### ISSUE TITLE
text for post as fixture fails

### ISSUE THREAD

**Issue by Rubyj** [jamielennox/requests-mock#96]
@jamielennox 
When my `text` to a post request is coming from a fixture, my POST request fails with the error:

```
Fixture "success_message" called directly. Fixtures are not meant to be called directly,
but are created automatically when test functions request them as parameters.
```

```
@pytest.fixture()
def mock_slack_url():
    return "http://postslack.com"


@pytest.fixture()
def success_message:
    return "success"


def test_post(mock_slack_url, requests_mock):
    requests_mock.post(mock_slack_url, text=success_message)
    response = requests.post(mock_slack_url, json=success_message)
    response.raise_for_status()
    assert response.text == success_message
```
Not sure if anyone is aware of this issue yet.

TLDR using a fixture in a `requests_mock.post(text=)` causes an error.

**Comment by jamielennox** [comment]
I think this is a problem of how you're using pytest. To use a pytest fixture you have to put it in the arguments of the function and have pytest call it for you. eg 
```
def test_post(mock_slack_url, requests_mock, success_message):
```
without having done this you're just passing a function to text= which means that `requests-mock` will execute it like a callback, thus the error.

**Comment by Rubyj** [comment]
Thank you for this tip.
